docs/docs/developers/contracts/references/portals/outbox.md

```diff
@@ -30,45 +34,33 @@ Allows a recipient to consume a message from the `Outbox`.
 
 | Name           | Type        | Description |
 | -------------- | -------     | ----------- |
-| `_message`     | `L2ToL1Msg` | The message to consume |
-| ReturnValue    | `bytes32`   | The hash of the message | 
+| `_l2BlockNumber`     | `uint256` | The block number specifying the block that contains the message we want to consume |
+| `_leafIndex`     | `uint256` | The index inside the merkle tree where the message is located |
+| `_message`     | `L2ToL1Msg` | The L2 to L1 message |
+| `_path`     | `bytes32[]` | The sibling path used to prove inclusion of the message, the _path length directly depends |
 
 #### Edge cases
 
-- Will revert with `Outbox__Unauthorized()` if `msg.sender != _message.recipient.actor`. 
+- Will revert with `Outbox__InvalidRecipient(address expected, address actual);` if `msg.sender != _message.recipient.actor`. 
 - Will revert with `Outbox__InvalidChainId()` if `block.chainid != _message.recipient.chainId`.
-- Will revert with `Outbox__NothingToConsume(bytes32 entryKey)` if the message does not exist.
-- Will revert with `Outbox__InvalidVersion(uint256 entry, uint256 message)` if the version of the entry and message sender don't match (wrong rollup).
+- Will revert with `Outbox__NothingToConsumeAtBlock(uint256 l2BlockNumber)` if the root for the block has not been set yet.
+- Will revert with `Outbox__AlreadyNullified(uint256 l2BlockNumber, uint256 leafIndex)` if the message at leafIndex for the block has already been consumed.
+- Will revert with `Outbox__InvalidPathLength(uint256 expected, uint256 actual)` if the existing height of the L2 to L1 message tree, and the supplied height do not match.
+- Otherwise, will revert with `MerkleLib__InvalidRoot(bytes32 expected, bytes32 actual)` if unable to verify the message existence in the tree.
```

Comment:
Don't have `otherwise` down here, makes it sound like it will always fail if there is options and "otherwise with this failure". Would just use the `will revert` start as the others 👍